client/new-stack-editor/lib/views/flexsplit/index.coffee

```diff
@@ -0,0 +1,89 @@
+kd = require 'kd'
+
+module.exports = class FlexSplit extends kd.View
+
+  @EVENT_EXPAND    = 'FlexSplit.EXPAND'
+  @EVENT_COLLAPSE  = 'FlexSplit.COLLAPSE'
+
+  @EVENT_RESIZED   = 'FlexSplit.RESIZED'
+  @EVENT_EXPANDED  = 'FlexSplit.EXPANDED'
+  @EVENT_COLLAPSED = 'FlexSplit.COLLAPSED'
+
+  @MAX = 100
+  @MIN = 0.0001
+
+  @HORIZONTAL =
+    name      : 'horizontal'
+    axis      : 'y'
+    getter    : 'getHeight'
+
+  @VERTICAL   =
+    name      : 'vertical'
+    axis      : 'x'
+    getter    : 'getWidth'
+
+
+  constructor: (options = {}, data) ->
+
+    options.cssClass   = kd.utils.curry 'flex-split', options.cssClass
+    options.sizes     ?= []
+    options.type      ?= FlexSplit.HORIZONTAL
+    options.resizable ?= yes
+
+    super options, data
+
+    @resizer = null
+    { @type, @name, storage } = @getOptions()
+    @setClass @type.name
+
+    @setupViews()
+
+    storage?.addView this, @name
+
+
+  createResizer: (view, size) ->
+
+    # This needs to be here to prevent circular dependency ~ GG
+    FlexSplitResizer = require './resizer'
+
+    @resizer = @addSubView new FlexSplitResizer { @type, view }
+
+    @forwardEvents @resizer, [
+      FlexSplit.EVENT_EXPANDED
+      FlexSplit.EVENT_RESIZED
+      FlexSplit.EVENT_COLLAPSED
+    ]
+
+    @resizer.on FlexSplit.EVENT_EXPAND, ->
+      if view.parent instanceof FlexSplit
+        view.parent.emit FlexSplit.EVENT_EXPAND
+
+    @resizer.on FlexSplit.EVENT_COLLAPSE, ->
+      if view.parent instanceof FlexSplit
```

Comment:
This would go to `constructor` not to the prototype of course.
```coffee
class FlexSplit extends kd.Object
  @_type = 'FlexSplit'
```

Than maybe `kd.Object` can export a function called `isInstance` that would help us to compare instances with their classes:

```coffee
split = new FlexSplit
kd.Object.isInstance split, 'FlexSplit'
# => true
```

Or maybe we can go one step further and make all the subclasses use this `is` function to check if an object is their instance:

```coffee
FlexSplit.isInstance split
# => true
```